### PR TITLE
Track multipart/resumable uploads GCS API calls

### PR DESCRIPTION
**Pull request by fcofdez**
Add tracking for multipart and resumable uploads for GoogleCloudStorage.
For resumable uploads only the last request is taken into account for
billing, so that's the only request that's tracked.

### CODE REVIEW

File: plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java [hunk 1]
```diff
@@ -289,17 +292,40 @@ protected boolean canFailRequest(final HttpExchange exchange) {
...
-        public void maybeTrack(final String request) {
+        public void maybeTrack(final String request, Headers requestHeaders) {
             if (Regex.simpleMatch("GET /storage/v1/b/*/o*", request)) {
                 trackRequest("LIST");
             } else if (Regex.simpleMatch("GET /download/storage/v1/b/*", request)) {
                 trackRequest("GET");
+            } else if (Regex.simpleMatch("PUT /upload/storage/v1/b/*", request) && isLastPart(requestHeaders)) {
+                trackRequest("PUT");
+            } else if (Regex.simpleMatch("POST /upload/storage/v1/b/*uploadType=multipart*", request)) {
+                trackRequest("PUT");
```
Comment (original-brownbear) [inline plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java:310]: Let's track this under a different key so we can identify the counts by type? Just `POST` is probably fine here as I currently can't think of another `POST` we do on GCS.

File: plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java [hunk 2]
```diff
@@ -335,6 +336,7 @@ private void writeBlobMultipart(BlobInfo blobInfo, InputStream inputStream, long
                 new Storage.BlobTargetOption[0];
             SocketAccess.doPrivilegedVoidIOException(
                     () -> client().create(blobInfo, buffer, targetOptions));
+            stats.trackPutObjectOperation();
```
Comment (original-brownbear) [inline plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java:339]: As mentioned below, let's give this a separate key to track under so we get metrics on what upload type we use how often.

**Review by original-brownbear — COMMENTED** [review]
Thanks! <= looks good, let's just count the different upload types separately

**Review by original-brownbear — APPROVED** [review]
LGTM

One thing I just thought of: maybe add a short comment to the `trackPostOperation` and `trackPutOperation` spots stating that we don't track this one the REST client layer because it would require complex inspection of the requests due to the multi-request nature of writes? (just to avoid future confusion about why we handle GETs and LISTs differently)

**Comment by fcofdez** [comment]
retest this please

**Pull request merged**